subprojects/core/src/integTest/groovy/org/gradle/api/tasks/TaskInputPropertiesIntegrationTest.groovy

```diff
@@ -417,31 +414,90 @@ apply from:'scriptPlugin.gradle'
             class MyTask extends DefaultTask {
                 @Input
                 FooType foo
+            }
+
+            task createFile(type: MyTask) {
+                ext.outputFile = file('output.txt')
+                outputs.file(outputFile)
+
+                foo = FooType.FOO
+
+                doLast {
+                    outputFile << foo
+                }
+            }
+"""
+
+        when:
+        succeeds 'createFile'
 
-                MyTask() {
-                    outputs.upToDateWhen {
-                        false
+        then:
+        executedTasks == [':createFile']
+        skippedTasks.empty
+
+        when:
+        succeeds 'createFile'
+
+        then:
+        executedTasks == [':createFile']
+        skippedTasks == [':createFile'] as Set
+    }
+
+    @Issue("gradle/gradle#919")
+    def "can use a custom Serializable type from init script as input property in this taskType: #taskType"() {
+        def initScript = file("init.gradle")
+
+        given:
+        initScript << """
+            enum FooType { FOO }
+
+            rootProject {
+                afterEvaluate {
+                    tasks.createFile {
+                        ext.fooType = FooType.FOO
+                        inputs.property('fooType', fooType)
                     }
                 }
             }
+"""
 
-            task neverUpToDate(type: MyTask) {
-                foo = FooType.FOO
+        buildFile << """
+            import org.gradle.api.internal.changedetection.state.InMemoryTaskArtifactCache
+
+            println "Flushing InMemoryTaskArtifactCache"
+            gradle.taskGraph.whenReady {
+                gradle.services.get(InMemoryTaskArtifactCache).invalidateAll()
```

Comment:
I tried appending code to the build script among other similar change but I couldn't reproduce what `InMemoryTaskArtifactCache.invalidateAll` does. In this case, I want to force Gradle to grab what is on disk and deserialize everything again in order to catch the issues where the wrong class loader is used. I copied over these line from [a previous test](https://github.com/gradle/gradle/pull/962/files#diff-96ce1abd714523fec969eaff46ff7accR350).

We could use `--no-daemon` flag, however, I wasn't able to make it work correctly. This issue will usually happen when the cache is flushed by the daemon which I haven't found a trivial deterministic way to execute.